### PR TITLE
Added handleTSIndexedAccessType.

### LINKED ISSUE #367: Handle TSIndexedAccessType
There currently isn't a handler for `TSIndexedAccessType`:

```ts
type ABC = {
  foo: string
}

type DEF = {
  bar: ABC["foo"] // <-- indexed type
}
```
The output for bar is `{ name: "unknown" }`. Expected output would be either:

`{ name: "string" }` or;
`{ name: "ABC[\"foo\"]" }`

I believe there is an ongoing discussion in #366 over which of these 2 it should be (based on whether ABC is declared as a type or an interface), but either way currently it is neither.

I'm happy to attempt a PR but I have zero experience with AST so perhaps only if this doesn't get picked up by someone more capable 😃

### PR DESCRIPTION
**Pull request by thchia**
Fixes #367

I'm new to working with ASTs, and welcome any feedback. I also had some issues with type narrowing in Flow, resulting in the use of `$Shape<>`.

### CODE REVIEW

File: src/utils/getTSType.js [hunk 1]
```diff
@@ -318,6 +320,40 @@ function handleTSTypeOperator(path: NodePath): ?FlowTypeDescriptor {
...
+    path.get('objectType'),
+    typeParams,
+  );
+  // eslint-disable-next-line no-undef
+  const indexType: $Shape<FlowLiteralType> = getTSTypeWithResolvedTypes(
+    path.get('indexType'),
+    typeParams,
+  );
+
+  // We only get the signature if the objectType is a type (vs interface)
```
Comment (devongovett) [inline src/utils/getTSType.js]: Why not for interfaces?

Comment (thchia) [reply src/utils/getTSType.js]: @devongovett actually I'm not 100% sure why, but I believe in my trial and error when defining the objectType as an interface, there is no signature property on it. I'm not sure if it is related to the discussion [here](https://github.com/reactjs/react-docgen/issues/366#issuecomment-495715639)?

**Comment by sylvesteraswin** [comment]
@thchia, thanks a lot for working on this. Whats the process on this PR?

**Comment by thchia** [comment]
@sylvesteraswin at this stage I’m still waiting on any additional comments from @devongovett :)

**Comment by danez** [comment]
I think we need to look at interfaces in a separate change. I think the change (for object types) makes sense.

**Pull request merged**